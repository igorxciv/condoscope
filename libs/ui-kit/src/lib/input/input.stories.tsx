import React, { FC } from 'react'
import { text, boolean, select } from '@storybook/addon-knobs'
import { Input } from './input'

export default {
  component: Input,
  title: 'Input',
}

export const regular: FC = () => {
  return (
    <Input
      placeholder={text('Placeholder', 'Placeholder')}
      disabled={boolean('Disabled', false)}
      mass={select('Size', {
        Small: 'small',
        Medium: 'medium',
        Large: 'large',
      }, 'medium')}
      appearance={select('Appearance', {
        Basic: 'basic',
        Primary: 'primary',
        Info: 'info',
        Success: 'success',
        Warning: 'warning',
        Danger: 'danger',
      }, 'basic')}
    >
      {text('Label', 'Label')}
    </Input>
  )
}

export const password: FC = () => {
  return (
    <Input placeholder={text('Placeholder', 'Password')} kind="password">
      {text('Label', 'Password')}
    </Input>
  )
}

export const number: FC = () => {
  return (
    <Input placeholder={text('Placeholder', '12345')} kind="number">
      {text('Label', 'Number')}
    </Input>
  )
}

export const email: FC = () => {
  return (
    <Input placeholder={text('Placeholder', 'email@me.com')} kind="email">
      {text('Label', 'E-mail')}
    </Input>
  )
}
