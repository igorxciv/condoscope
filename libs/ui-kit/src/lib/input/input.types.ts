export type InputProps = {
  kind?: 'text' | 'number' | 'email' | 'password'
  // size?: 'small' | 'medium' | 'large'
  appearance?: 'basic' | 'success' | 'info' | 'warning' | 'error'
  placeholder?: string
  disabled?: boolean
}
