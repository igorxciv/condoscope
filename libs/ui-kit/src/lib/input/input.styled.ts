import styled, { css } from 'styled-components'
import { math, rem } from 'polished'
import { InputProps } from './input.types'
import { propertyAnimation, visuallyHidden } from '../__design/utils'

const basic = css<InputProps>`
  ${({ theme }) => css`
    --border-color: ${theme['input-basic-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-focus-border-color']};
    }
  `}
`

const info = css`
  ${({ theme }) => css`
    --border-color: ${theme['input-info-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-info-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-info-focus-border-color']};
    }
  `}
`

const primary = css`
  ${({ theme }) => css`
    --border-color: ${theme['input-primary-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-primary-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-primary-focus-border-color']};
    }
  `}
`

const success = css`
  ${({ theme }) => css`
    --border-color: ${theme['input-success-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-success-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-success-focus-border-color']};
    }
  `}
`

const warning = css`
  ${({ theme }) => css`
    --border-color: ${theme['input-warning-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-warning-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-warning-focus-border-color']};
    }
  `}
`

const danger = css`
  ${({ theme }) => css`
    --border-color: ${theme['input-danger-border-color']};

    &:not(:disabled):hover {
      --border-color: ${theme['input-danger-hover-border-color']};
    }

    &:focus {
      --border-color: ${theme['input-danger-focus-border-color']};
    }
  `}
`

const inputAppearances = { basic, info, primary, success, warning, danger }

export const InputGroupStyled = styled.div<InputProps>`
  display: grid;
  row-gap: ${rem(8)};
  grid-template-columns: min-content min-content;

  ${({ theme }) => css`
    font-family: ${theme['font-family-primary']};
  `}
`

export const InputLabelStyled = styled.label`
  text-transform: uppercase;

  &[aria-hidden="true"] {
      ${visuallyHidden()}
  }

  ${({ theme }) => css`
    color: ${theme['input-label-color']};
    font-size: ${theme['font-size-label']};
    font-weight: ${theme['font-weight-bold']};
  `}
`

export const InputHelpTextStyled = styled.span`
  ${({ theme }) => css`
    color: ${theme['input-label-color']};
    font-size: ${theme['font-size-caption']};
    align-self: flex-end;
    justify-self: flex-end;
  `}
`

export const InputFieldGroupStyled = styled.div<InputProps>`
  position: relative;
  grid-column: 1 / 3;

  ${({ theme, mass, appearance }) => css`
    svg {
      width: ${theme['input-icon-size']};
      height: ${theme['input-icon-size']};
      position: absolute;
      right: ${theme['input-horizontal-padding']};
      bottom: ${theme[`input-${mass}-vertical-padding`]};
      color: ${theme[`input-icon-${appearance}-color`]};
    }
  `}
`

export const InputFieldStyled = styled.input<InputProps>`
  ${({ theme, appearance, mass, inputIcon }) => css`
    --background-color: ${theme['input-background']};

    border-radius: ${theme['input-radius']};
    border: ${rem(1)} solid var(--border-color);

    line-height: ${theme['input-line-height']};

    font-size: ${theme['font-size-p1']};
    font-weight: ${theme['font-weight-semi-bold']};
    color: ${theme['input-basic-color']};
    background-color: var(--background-color);

    padding: ${theme[`input-${mass}-vertical-padding`]} ${theme['input-horizontal-padding']};

    ${inputIcon && css`
      padding-right: ${math(`${theme['input-horizontal-padding']} + ${theme['input-icon-size']}`)};
    `}

    ${propertyAnimation(['background-color', 'border-color'])}

    &::placeholder {
      color: ${theme['input-placeholder-color']};
      font-weight: ${theme['font-weight-regular']};
      font-family: ${theme['font-family-primary']};
    }

    &:disabled,
    &:disabled::placeholder {
      color: ${theme['input-disabled-color']};
    }

    &:focus {
      --background-color: ${theme['input-focus-background']};
      outline: none;
    }

    &:not(:disabled):hover {
      --background-color: ${theme['input-hover-background']};
    }

    ${inputAppearances[appearance]}
  `}
`

export const InputCaptionStyled = styled.span<InputProps>`
  grid-column: 1 / 3;

  ${({ theme, appearance }) => css`
    color: ${theme[`input-caption-${appearance}-color`]};
    font-size: ${theme['font-size-caption']};
  `}
`
