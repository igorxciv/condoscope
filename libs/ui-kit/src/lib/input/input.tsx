import React, { FC, PropsWithChildren } from 'react'
import { InputGroupStyled, InputLabelStyled, InputHelpTextStyled, InputFieldGroupStyled, InputFieldStyled, InputCaptionStyled } from './input.styled'
import { InputProps } from './input.types'

export const Input: FC<InputProps> = (props: PropsWithChildren<InputProps>) => {
  const { children, inputIcon, kind, caption: captionText, appearance, mass, disabled } = props
  const label = children ? (
    <InputLabelStyled>
      {children}
    </InputLabelStyled>
  ) : (
    <InputLabelStyled aria-hidden="true">
      Empty Label
    </InputLabelStyled>
  )
  const caption = captionText ? (
    <InputCaptionStyled appearance={appearance}>
      {captionText}
    </InputCaptionStyled>
  ) : null

  return (
    <InputGroupStyled>
      {label}
      <InputHelpTextStyled>
        Help
      </InputHelpTextStyled>

      <InputFieldGroupStyled mass={mass} appearance={appearance} disabled={disabled}>
        <InputFieldStyled type={kind} appearance={appearance} mass={mass} inputIcon={inputIcon} />
        {inputIcon}
      </InputFieldGroupStyled>

      {caption}
    </InputGroupStyled>
  )
}

Input.defaultProps = {
  kind: 'text',
  appearance: 'basic',
  placeholder: '',
  disabled: false,
  mass: 'medium',
}
