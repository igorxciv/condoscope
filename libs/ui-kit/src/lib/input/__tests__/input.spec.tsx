import React from 'react'
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow'
import renderer from 'react-test-renderer'
import { Input } from '../input'
import 'jest-styled-components'

describe('Input', () => {
  let shallowRenderer: ShallowRenderer
  let component

  beforeEach(() => {
    shallowRenderer = createRenderer()
    component = shallowRenderer.render(
      <Input>
        Input
      </Input>,
    )
  })

  test('renders successfully', () => {
    expect(component).toMatchSnapshot()
  })

  describe('when no children provided', () => {
    beforeEach(() => {
      component = shallowRenderer.render(<Input />)
    })

    test('renders successfully', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('when has caption', () => {
    beforeEach(() => {
      component = shallowRenderer.render(
        <Input caption="Caption">
          Input
        </Input>,
      )
    })

    test('renders with caption', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('when disabled', () => {
    beforeEach(() => {
      component = shallowRenderer.render(
        <Input disabled>
          Input
        </Input>,
      )
    })

    it('renders successfully', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('when has icon', () => {
    beforeEach(() => {
      component = shallowRenderer.render(
        <Input icon={
          <span>
            icon
          </span>
        }
        >
          Input
        </Input>,
      )
    })

    test('renders successfully', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('full renderer', () => {

    describe('Primary', () => {

      beforeEach(() => {
        component = renderer.create(<Input appearance="primary" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })

    describe('Info', () => {
      beforeEach(() => {
        component = renderer.create(<Input appearance="info" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })

    describe('Basic', () => {
      beforeEach(() => {
        component = renderer.create(<Input appearance="basic" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })

    describe('Success', () => {
      beforeEach(() => {
        component = renderer.create(<Input appearance="success" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })

    describe('Warning', () => {
      beforeEach(() => {
        component = renderer.create(<Input appearance="warning" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })

    describe('Danger', () => {
      beforeEach(() => {
        component = renderer.create(<Input appearance="danger" />)
      })

      test('renders successfully', () => {
        expect(component).toMatchSnapshot()
      })
    })
  })
})
