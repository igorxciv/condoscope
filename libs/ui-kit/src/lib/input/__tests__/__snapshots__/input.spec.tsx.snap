// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input full renderer Basic renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input full renderer Danger renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input full renderer Info renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input full renderer Primary renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input full renderer Success renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input full renderer Warning renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    Input
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Input when disabled renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    Input
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={true}
  >
    <input
      className="c4"
      disabled={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input when has caption renders with caption 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c5 {
  grid-column: 1 / 3;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    Input
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
  <span
    className="c5"
  >
    Caption
  </span>
</div>
`;

exports[`Input when no children provided renders successfully 1`] = `
.c0 {
  display: grid;
  row-gap: 0.5rem;
  grid-template-columns: auto auto;
}

.c1 {
  text-transform: uppercase;
}

.c1[aria-hidden="true"] {
  border: none;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

.c3 {
  position: relative;
  grid-column: 1 / 3;
}

.c3 svg {
  position: absolute;
}

.c4 {
  border: 0.0625rem solid var(--border-color);
  width: 100%;
  box-sizing: border-box;
  background-color: var(--background-color);
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color border-color;
  transition-property: background-color border-color;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
}

<div
  className="c0"
>
  <label
    aria-hidden="true"
    className="c1"
  >
    Empty Label
  </label>
  <span
    className="c2"
  >
    Help
  </span>
  <div
    className="c3"
    disabled={false}
  >
    <input
      className="c4"
      disabled={false}
      type="text"
    />
  </div>
</div>
`;
